src/hypothesis/strategies.py

```diff
@@ -454,11 +454,33 @@ def fractions():
     )
 
 
+_special_floats = {
+    float('inf'): Decimal('Infinity'),
+    float('-inf'): Decimal('-Infinity'),
+}
+
+
+def float_to_decimal(f):
```

Comment:
This should go in hypothesis.internal.compat, which is where all the weird stuff for working across versions goes, and it should be definition guarded behind a version check and otherwise just call decimal(float).